src/bootstrap/tool.rs

```diff
@@ -853,7 +854,18 @@ macro_rules! tool_extended {
                     } else {
                         SourceType::Submodule
                     },
-                })
+                })?;
+
+                if (false $(|| $add_to_sysroot)?) && $sel.compiler.stage > 0 {
+                    let bin_dir = $builder.sysroot($sel.compiler).join("bin");
+                    t!(fs::create_dir_all(&bin_dir));
+                    let bin_tool = bin_dir.join(exe($tool_name, $sel.compiler.host));
+                    let _ = fs::remove_file(&bin_tool);
+                    $builder.copy(&tool, &bin_tool);
+                    Some(bin_tool)
```

Comment:
hmm, why is the path we return from the Step different depending on whether it's in the sysroot or not? I would expect this to be consistent with ToolBuild even if we copy the tool to the sysroot as a side effect.

Reply:
We do this for rustdoc for some reason: https://github.com/rust-lang/rust/blob/503e19d01e941b88bf6d5b28e9108d046abcfa2d/src/bootstrap/tool.rs#L567-L576

I copied this from there

Reply:
Fascinating. ok, seems good to be consistent.